src/snowflake/snowpark/async_job.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+from snowflake.snowpark._internal.server_connection import SnowflakeConnection
+
+
+class AsyncJob:
+    def __init__(
+        self,
+        query_id: str = None,
```

Comment:
can it be none? If not, remove default value. And the same for query and conn, so you don't need to any check in the function.